### PR TITLE
Created documentation of "Chords"

### PR DESCRIPTION
**Pull request by Ritika8081**
@lorforlinux I have added the documentation for the new application "Chords" to provide clarity and enhance understanding.

### CODE REVIEW

File: software/chords/index.rst [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+#################
+
+Overview
+********
+Chords is an open source web application designed for real-time signal visualization, particularly tailored for bio-potential signals like EEG, EMG, ECG and EOG. This tool serves as an advanced alternative to the standard Arduino IDE serial plotter, offering enhanced functionality for students, researchers and hobbyists alike who want to record biopotential data using  `BioAmp hardware <https://docs.upsidedownlabs.tech/hardware/index.html>`_.
+
+.. figure:: media/chords_landing_page.*
+    :align: center
+    :alt: chords landing page
+
```
Comment (lorforlinux) [inline software/chords/index.rst:13]:
```suggestion
.. figure:: media/chords_landing_page.*
    :align: center
    :alt: chords landing page
    
    chords landing page

```

File: software/chords/index.rst [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+- **Recording**: You can record data in CSV format indefinitely or set a timer to stop recording automatically when it expires.
+- **Download/Delete**: Easily download or delete recorded data with just a few clicks. Manage your data efficiently, allowing for quick access to relevant data or removal of unwanted data to free up storage.
+- **Zoom**: Zoom-in and zoom-out feature, gives you more control as now you can either zoom in to look at signal or zoom out for an overall view.
+- **Mode Toggle**: Offers both light and dark modes, enabling you to switch according to your preference for a more comfortable user experience.
+- **Pause/Play**: Includes options to freeze and resume data display, allowing users to pause the visualization for detailed examination and then continue data monitoring seamlessly.
+- **Channel**: Supports plotting of up to six channels of data simultaneously in real time where each stream is color coded differently for easy identification.
+- **Disconnect**: Terminate the connection to the development board with a simple click. This functionality ensures a hassle-free disconnection process when you’re done with data collection or visualization.
+
```
Comment (lorforlinux) [inline software/chords/index.rst:25]:
```suggestion
Features
********
- **Connection**: It is effortlessly compatible with BioAmp Hardware attached to an Arduino-compatible development board running the `Chords Arduino firmware <https://github.com/upsidedownlabs/Chords-Arduino-Firmware>`_, simplifying the setup process and ensuring a smooth and efficient workflow from data acquisition to visualization.
- **Real-time Visualization**: Experience smooth real-time visualization of incoming data rendered with `WebGL-Plot <https://github.com/danchitnis/webgl-plot>`_. The system ensures smooth signal display, helping you monitor the flow without interruptions or lag.
- **Recording**: You can record data in CSV format indefinitely or set a timer to stop recording automatically when it expires.
- **Download/Delete**: Easily download or delete recorded data with just a few clicks. Manage your data efficiently, allowing for quick access to relevant data or removal of unwanted data to free up storage.
- **Zoom**: Zoom-in and zoom-out feature, gives you more control as now you can either zoom in to look at signal or zoom out for an overall view.
- **Theme Toggle**: Offers both light and dark themes, enabling you to switch according to your preference for a more comfortable user experience.
- **Pause/Play**: Includes options to freeze and resume data display, allowing users to pause the visualization for detailed examination and then continue data monitoring seamlessly.
- **Channel**: Supports plotting of up to six channels of data simultaneously in real time where each stream is color-coded differently for easy identification.
- **Disconnect**: Terminate the connection to the development board with a simple click. This functionality ensures a hassle-free disconnection process when you’re done with data collection or visualization.

```

File: software/chords/index.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+Software Requirements
+*********************
+
+* You have to install `Chords Arduino Firmware <https://github.com/upsidedownlabs/Chords-Arduino-Firmware>`_ using arduino IDE.
+* For now Chords is compatible with chromium based browsers. To ensure compatibility with supported browsers, see :ref:`chords-browsers-compatibility`.
+
```
Comment (lorforlinux) [inline software/chords/index.rst:32]:
```suggestion
Software Requirements
*********************

* You have to flash  `Chords Arduino Firmware <https://github.com/upsidedownlabs/Chords-Arduino-Firmware>`_ to your Arduino board using Arduino IDE.
* For now the Chords web application is only compatible with Chromium-based browsers. To check your browser compatibility, see :ref:`chords-browsers-compatibility` section of this page.

```

File: software/chords/index.rst [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+Uploading the Code
+------------------
+
+1. Connect the Arduino board to your laptop using the USB cable.
+2. Copy and paste the `Arduino Firmware  <https://github.com/upsidedownlabs/Chords-Arduino-Firmware>`_ into the Arduino IDE.
+3. Select your board under `Board > Arduino AVR` in the tools menu.
+4. Choose the correct COM port in the tools menu (the one that disappears when you disconnect the board).
+5. Upload the code and open Chords in your web browser.
+
```
Comment (lorforlinux) [inline software/chords/index.rst:59]:
```suggestion
How to Use Chords
*******************

Uploading the Code
------------------

1. Connect the Arduino board to your laptop using the USB cable.
2. Copy and paste the `Arduino Firmware  <https://github.com/upsidedownlabs/Chords-Arduino-Firmware>`_ into the Arduino IDE.
3. Select your board from `Board` section in the tools menu.
4. Choose the correct COM port in the tools menu (the one that disappears when you disconnect the board).
5. Click on the `Upload` button to flash your Arduino board.
```

File: software/chords/index.rst [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+- A USB cable (type depends on board)
+- BioAmp Hardware : 
+
+  - `BioAmp EXG Pill <https://docs.upsidedownlabs.tech/hardware/bioamp/bioamp-exg-pill/index.html>`_
+  - `Muscle BioAmp Biscute <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-biscute/index.html>`_
+  - `Muscle BioAmp Candy <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-candy/index.html>`_
+  - `Muscle BioAmp Patchy <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-patchy/index.html>`_
+  - `Muscle BioAmp Shield <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-shield/index.html>`_
+  
+
```
Comment (lorforlinux) [inline software/chords/index.rst:47]:
```suggestion
Hardware Requirements
*********************
To use Chords, you need:

- Compatible Development Board : Arduino Uno R3 And R4 Minima
- A USB cable (type depends on board)
- BioAmp Hardware : 

  - `BioAmp EXG Pill <https://docs.upsidedownlabs.tech/hardware/bioamp/bioamp-exg-pill/index.html>`_
  - `Muscle BioAmp Biscute <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-biscute/index.html>`_
  - `Muscle BioAmp Candy <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-candy/index.html>`_
  - `Muscle BioAmp Patchy <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-patchy/index.html>`_
  - `Muscle BioAmp Shield <https://docs.upsidedownlabs.tech/hardware/bioamp/muscle-bioamp-shield/index.html>`_
  
```

**Comment by lorforlinux** [comment]
@Ritika8081 I have suggested changes, please check carefully and then apply them. You should pull these changes locally and check if everything is building fine or now. If it looks good on local build then I will merge this PR.

**Comment by Ritika8081** [comment]
@lorforlinux  it's working fine for me and I have removed extra spaces.

File: software/chords/index.rst (hunk 5 above)
Comment (Ritika8081) [reply software/chords/index.rst:49]: no change suggested, resolving this conversation.

File: software/chords/index.rst [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+- **Download/Delete**: Easily download or delete recorded data with just a few clicks. Manage your data efficiently, allowing for quick access to relevant data or removal of unwanted data to free up storage.
+- **Zoom**: Zoom-in and zoom-out feature, gives you more control as now you can either zoom in to look at signal or zoom out for an overall view.
+- **Theme Toggle**: Offers both light and dark themes, enabling you to switch according to your preference for a more comfortable user experience.
+- **Pause/Play**: Includes options to freeze and resume data display, allowing users to pause the visualization for detailed examination and then continue data monitoring seamlessly.
+- **Channel**: Supports plotting of up to six channels of data simultaneously in real time where each stream is color-coded differently for easy identification.
+- **Disconnect**: Terminate the connection to the development board with a simple click. This functionality ensures a hassle-free disconnection process when you’re done with data collection or visualization.
+
+
+Software Requirements
+*********************
```
Comment (lorforlinux) [inline software/chords/index.rst:30]: We have extra blank line here.

File: software/chords/index.rst [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+- **Microsoft Edge**
+- **Opera**
+
+If the user's browser does not support the Web Serial API, a message will inform them of the incompatibility, recommending the use of a supported browser.
+
+For more information, refer to MDN Web Docs on the `Web Serial API <https://developer.mozilla.org/en-US/docs/Web/API/Web_Serial_API>`_.
+
+
+Technologies Used
+*******************
```
Comment (lorforlinux) [inline software/chords/index.rst:175]: Why do we have these extra blank lines?

File: software/chords/index.rst [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+- **Tailwind CSS**: Utility-first CSS framework.
+- **Shadcn UI**: Reusable and customizable components, built with Radix UI.
+- **WebGL Plot**: Real-time charting library.
+- **IndexedDB API**: Browser-based database for storing recorded data.
+
+.. figure:: media/chords_tech_stack.*
+    :align: center
+    :alt: chords tech stack
+
+
```
Comment (lorforlinux) [inline software/chords/index.rst:187]: We have to provide the title to every figure, I did the same for one other image above doing it again here. Please make sure to do it for other independent figures. You can only skip it for the figures you have put in that table.

```suggestion
    :align: center
    :alt: chords tech stack
    
    chords tech stack
```

Comment (Ritika8081) [reply software/chords/index.rst:187]: okay

**Comment by lorforlinux** [comment]
@Ritika8081 Thank you so much for your contribution, I am merging this now.

**Pull request merged**